include/dict.hpp

```diff
@@ -99,9 +99,11 @@ constexpr all_nodes_t visit_all_nodes;
 template<typename... Types>
 struct Dict
 {
-    using node_ptr = std::shared_ptr<Dict>;
-    using map_t    = std::map<std::string, node_ptr>;
-    using data_t   = std::variant<NoValue, map_t, Types...>;
+    using node_ptr     = std::shared_ptr<Dict>;
+    using map_t        = std::map<std::string, node_ptr>;
+    using data_t       = std::variant<NoValue, map_t, Types...>;
+    using empty_leaf_t = NoValue;
```

Comment:
@Dekken @nicolasaunai I've added this to deprecate use of map_t and NoValue from client code.
-> both come from Dict class and names reflect more what they are from client code.

Reply:
should we drop NoValue and use "monostate" 

https://en.cppreference.com/w/cpp/utility/variant/monostate

Reply:
hmm, good question, maybe, we can still alias monostate in Dict class to make change transparent.
```
using empty_leaf_t = std::monostate;
```